### PR TITLE
Next theme

### LINKED ISSUE #64: LitElement conversion part 3: theme and visual tests
### Convert theme files

Use following file structure:

- `vaadin-progress-bar-css.ts`

```ts
import { css } from 'lit-element';
import '@vaadin/vaadin-lumo-styles/color.js';
import '@vaadin/vaadin-lumo-styles/sizing.js';
import '@vaadin/vaadin-lumo-styles/spacing.js';
import '@vaadin/vaadin-lumo-styles/style.js';

export const progressBarStyles = css`
  /* styles go here */
`;
```

- `vaadin-progress-bar-styles.ts`

```ts
import { registerStyles } from '@vaadin/vaadin-themable-mixin/register-styles.js';
import { progressBarStyles } from './vaadin-split-layout-css';

registerStyles('vaadin-progress-bar', componentStyles, { moduleId: 'lumo-progress-bar' });
```

- `vaadin-progress-bar.ts`

```ts
import './vaadin-progress-bar-styles';
import '../../src/vaadin-progress-bar';
```

For Material, remember to change `moduleId` and use imports from `@vaadin/vaadin-material-styles`.

**Note**: remember to add `theme` folder back to ESLint script.

### Convert visual tests

- Update `.gemini.yml` to use new `rootUrl`, plugin `esm` and Chrome 69+ 

- Restore `npm run test:visual` command in `.travis.yml` for Travis

- Remove Firefox screenshots (those are for old Firefox with ShadyDOM)

- Update `test/visual/test.js` file to align with the skeleton:
  - remove `goToAboutBlank` workaround (no longer needed)

  - **important**: remove leading slashes `/` in `suite.setUrl`

  - use arrow functions to get rid of ESLint warnings

- Update actual visual tests to use ES modules imports

  - [`gemini-esm`](https://github.com/web-padawan/gemini-esm) is using `es-dev-server` with compatbility set to "none"
  
  - `async` / `await` and any other ES2018 features are safe to use
  
  - remove webcomponents polyfills from the tests completely
  
  - replace `document.write` usage with dynamic imports

- Run `npm run gemini` to update visual tests screenshots

### PR DESCRIPTION
**Pull request by phanfullstack-12345**
Fixes #64

### CODE REVIEW

File: test/visual/default.html [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
+    <meta charset="UTF-8">
+    <title>Progress bar default tests</title>
+  
+    <script>
+      window.addEventListener('WebComponentsReady', function() {
+        customElements.whenDefined('vaadin-progress-bar').then(function() {
+          window.ShadyDOM && window.ShadyDOM.flush(); // Force DOM composition
+          window.webComponentsAreReady = true; // Checked in gemini before capture callback
+        });
       });
```
Comment (web-padawan) [inline test/visual/default.html]: Please remove this block, it does nothing. 
The `WebComponentsReady` event is dispatched by polyfill and now we removed it.

File: test/visual/rtl.html [hunk 2]
```diff
@@ -1,28 +1,41 @@
...
-    window.addEventListener('WebComponentsReady', function() {
-      customElements.whenDefined('vaadin-progress-bar').then(function() {
-        window.ShadyDOM && window.ShadyDOM.flush(); // Force DOM composition
-        window.webComponentsAreReady = true; // Checked in gemini before capture callback
+      window.addEventListener('WebComponentsReady', function() {
+        customElements.whenDefined('vaadin-progress-bar').then(function() {
+          window.ShadyDOM && window.ShadyDOM.flush(); // Force DOM composition
+          window.webComponentsAreReady = true; // Checked in gemini before capture callback
+        });
       });
```
Comment (web-padawan) [inline test/visual/rtl.html]: Please remove this block, it does nothing. 
The `WebComponentsReady` event is dispatched by polyfill and now we removed it.

File: test/visual/rtl.html [hunk 3]
```diff
@@ -1,28 +1,41 @@
...
+      <vaadin-progress-bar value="0.1">
+      </vaadin-progress-bar>
+    </span>
+  
+    <script type="module">
+      (async() => {
+        const progressBar = document.querySelector('vaadin-progress-bar');
+        const theme = window.location.search.replace(/.*theme=(\w+).*/, '$1') || 'lumo';
+  
+        await import('../../theme/' + theme + '/vaadin-progress-bar.html');
```
Comment (web-padawan) [inline test/visual/rtl.html]: Should be `.js` extension.

File: test/visual/default.html [hunk 4]
```diff
@@ -1,27 +1,40 @@
...
+      <vaadin-progress-bar value="0.1">
+      </vaadin-progress-bar>
+    </span>
+  
+    <script type="module">
+      (async() => {
+        const progressBar = document.querySelector('vaadin-progress-bar');
+        const theme = window.location.search.replace(/.*theme=(\w+).*/, '$1') || 'lumo';
+  
+        await import('../../theme/' + theme + '/vaadin-progress-bar.html');
```
Comment (web-padawan) [inline test/visual/default.html]: Should be `.js` extension.

Comment (web-padawan) [inline test/visual/default.html]: Should be `.js` extension.

**Review by web-padawan — CHANGES_REQUESTED** [review]

**Review by web-padawan — CHANGES_REQUESTED** [review]
Mostly LGTM, pushed a fix to get rid of `stylelint` errors.

I also have two more tasks, please do them also in this PR:

### 1. Lumo variants test 

Please add `lumo-variants.html` visual test file with progress bars that have `theme`:
  - `theme="contrast"`
  - `theme="sucess"`
  - `theme="error"`

This file should be only tested for Lumo, not Material. 
So you can import progress bar there with usual import (instead of dynamc import):
```js
import '../../theme/lumo/vaadin-progress-bar.js';
```

See [example](https://github.com/vaadin/vaadin-split-layout/blob/6b70d8149f5392583ac866a1c237e4ac2bace6b4/test/visual/test.js#L26) in `vaadin-split-layout`.

### 2. RTL test update

Currently, `default.html` and `rtl.html` tests code is almost the same, except `dir` attribute.
Let's change this to only use one file (`default.html`) and set `dir` based on the URL parameter.

See [example](https://github.com/vaadin/vaadin-radio-button/blob/1b274fba6d59238876e25c43e8a0400811a92e9b/test/visual/test.js#L15-L20) in `vaadin-radio-button`.

File: test/visual/test.js [hunk 5]
```diff
@@ -12,14 +12,20 @@ gemini.suite('vaadin-progress-bar', rootSuite => {
       suite
         .setUrl(`default.html?theme=${theme}`)
         .setCaptureElements('#default-tests')
-        .capture('vaadin-progress-bar');
+        .capture('vaadin-progress-bar')
```
Comment (web-padawan) [inline test/visual/test.js]: This `capture` creates another folder with the same screenshots as in `default` (the next line).

Please remove this line, and the corresponding folders with screenshots it creates:

```
vaadin-progress-bar/default-tests-lumo/vaadin-progress-bar/chrome.png
vaadin-progress-bar/default-tests-material/vaadin-progress-bar/chrome.png
```

**Review by web-padawan — CHANGES_REQUESTED** [review]

File: test/visual/test.js (hunk 5 above)
Comment (phanfullstack-12345) [reply test/visual/test.js]: done

**Review by web-padawan — APPROVED** [review]

**Pull request merged**